src/main/scala/org/broadinstitute/hail/variant/VariantSampleMatrix.scala

```diff
@@ -218,21 +218,26 @@ case class VSMSubgen[T](
   tGen: (Variant) => Gen[T]) {
 
   def gen(sc: SparkContext)(implicit tct: ClassTag[T]): Gen[VariantSampleMatrix[T]] =
-    for (subsizes <- Gen.getPartition(7);
+    for (subsizes <- Gen.getPartition(6);
       vaSig <- vaSigGen.resize(subsizes(0));
       saSig <- saSigGen.resize(subsizes(1));
       globalSig <- globalSigGen.resize(subsizes(2));
-      sampleIds <- sampleIdGen.resize(subsizes(3));
-      nSamples = sampleIds.length;
       global <- globalGen(globalSig).resize(subsizes(4));
+      nPartitions <- Gen.getRng().map(_.nextInt(1,10));
```

Comment:
Yes definitely. I ought to commit to memory the operations already present in Gen. This also eliminates the need for `getRng` though I wonder if it'd be easier to re-expose `RandomDataGenerator` through something like:

```scala
def rand[T](m: RandomDataGenerator => T): Gen[T] =
  Gen { p => m(p.rng) }
```